### PR TITLE
Adding release process documentation

### PR DESCRIPTION
**Pull request by pvegh-oc**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,3 +46,23 @@ provider "maas" {
 ### Resources and Data Sources Configuration
 
 The [docs](/docs) section contains details about each supported Terraform resource and data source.
+
+### Release process
+
+1. Checkout main and create a new branch as `release-vX.X.X`
+   ```bash
+   git checkout main
```
Comment (sparkiegeek) [inline README.md:54]: Hmm, branch here is still called 'master'

We'd like to change it but don't want to break all your clones

Comment (pvegh-oc) [reply README.md]: My mistake, fixed it.

**Review by greg-oc — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -46,3 +46,23 @@ provider "maas" {
...
 The [docs](/docs) section contains details about each supported Terraform resource and data source.
+
+### Release process
+
+1. Checkout master and create a new branch as `release-vX.X.X`
+   ```bash
+   git checkout master  
+   git checkout -b release-vX.X.X
+   git push -u origin release-vX.X.X
+   ```
```
Comment (sparkiegeek) [inline README.md:57]:
```suggestion
1. Create a new branch from `master` as `release-vX.X.X`
   ```bash
   git branch release-vX.X.X master
   git push -u origin release-vX.X.X
   ```

```

File: README.md [hunk 3]
```diff
@@ -46,3 +46,23 @@ provider "maas" {
...
+2. Raise a PR on github, title of the PR should be in the following format
+   `Release vX.X.X`
+3. Update the `CHANGELOG.md` with your release version, date and change details.
+4. Go to [Releases](https://github.com/maas/terraform-provider-maas/releases) over on github
+5. Click [Draft a new release](https://github.com/maas/terraform-provider-maas/releases/new)
+6. On `Target` choose the lastest commit you want to release for
+7. Set the `release title` to the release version, for example `v1.0.1`
+8. Copy and paste the relevant CHNAGELOG.md entries to the release description
```
Comment (sparkiegeek) [inline README.md:65]:
```suggestion
8. Copy and paste the relevant CHANGELOG.md entries to the release description
```

File: README.md [hunk 4]
```diff
@@ -46,3 +46,23 @@ provider "maas" {
...
```
[the remark this answers is not in the document]
Comment (pvegh-oc) [reply README.md]: Thanks, fixed the typo.

**Review by greg-oc — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (pvegh-oc) [reply README.md]: Rephrased the sentence as suggested. Thanks

Comment (sparkiegeek) [reply README.md]: You missed the changes to the git commands?

I supplied different commands which avoid you checking out two branches (master, then the release branch) when all you need to do is create the branch and push it to GitHub.

Your version will leave your local checkout on the release which is an unlikely location to be wanting to make changes :shrug:

Comment (pvegh-oc) [reply README.md]: Ok, I will update them. 
It is just usually how I create a new branch, off from a base. 
`git checkout -b` is for creating a new branch.

File: README.md [hunk 5]
```diff
@@ -46,3 +46,22 @@ provider "maas" {
...
+   ```bash
+   git branch release-vX.X.X master
+   git push -u origin release-vX.X.X
+   ```
+2. Raise a PR on github, title of the PR should be in the following format
+   `Release vX.X.X`
+3. Update the `CHANGELOG.md` with your release version, date and change details.
+4. Go to [Releases](https://github.com/maas/terraform-provider-maas/releases) over on github
+5. Click [Draft a new release](https://github.com/maas/terraform-provider-maas/releases/new)
+6. On `Target` choose the lastest commit you want to release for
```
Comment (sparkiegeek) [inline README.md:62]:
```suggestion
6. On `Target` choose the latest commit you want to release for
```

**Review by sparkiegeek — APPROVED** [review]

**Pull request merged**